ts/features/walletV3/payment/screens/WalletPaymentPickMethodScreen.tsx

```diff
@@ -44,51 +83,238 @@ const WalletPaymentPickMethodScreen = () => {
-      </GradientScrollView>
-    </BaseScreenComponent>
+    <GradientScrollView
+      primaryActionProps={{
+        label: I18n.t("global.buttons.continue"),
+        accessibilityLabel: I18n.t("global.buttons.continue"),
+        onPress: handleContinue,
+        disabled: isLoading || !canContinue,
+        loading: isLoading
+      }}
+    >
+      <H2>{I18n.t("wallet.methodSelection.header")}</H2>
+      <VSpacer size={16} />
+      {shouldShowWarningBanner && (
+        <Alert
+          content={I18n.t("wallet.methodSelection.alert.body")}
+          variant="warning"
+          viewRef={alertRef}
+          onPress={() => null}
+          action={I18n.t("wallet.methodSelection.alert.cta")}
+        />
+      )}
+      <ListItemHeader label={I18n.t("wallet.methodSelection.yourMethods")} />
+
+      <RadioGroup<string>
+        selectedItem={selectedMethod?.walletId}
+        items={savedMethodsListItems}
+        onPress={handleSelectSavedMethod}
+      />
+
+      <ListItemHeader label={I18n.t("wallet.methodSelection.otherMethods")} />
+
+      <RadioGroup<string>
+        items={notSavedMethodsListItems}
+        selectedItem={selectedMethod?.methodId}
+        onPress={handleSelectNotSavedMethod}
+      />
+    </GradientScrollView>
   );
 };
 
+// ----------------------- UTILS -----------------------
+
+const getIconWithFallback = (
+  brand?: string
+): ComponentProps<typeof ListItemRadio>["startImage"] => {
+  const logos = IOPaymentLogos;
+  return pipe(
+    brand,
+    O.fromNullable,
+    O.chain(findFirstCaseInsensitive(logos)),
+    O.map(([brand]) => brand),
+    O.fold(
+      () => ({ icon: "creditCard" }),
+      // @ts-expect-error ts whines because this function can return
+      // two different -- both correct -- types (see return type)
+      brand => ({ paymentLogo: brand as IOPaymentLogos })
+    )
+  );
+};
+const mapGenericToRadioItem = (
+  method: PaymentMethodResponse,
+  transactionAmount?: number
+): RadioItem<string> => ({
+  id: method.id,
+  value: method.name,
+  disabled: isDisabled(method, transactionAmount),
+  startImage: getIconWithFallback(method.asset)
+});
+
+// should never return void, but since this is a map function it's expectable
+const mapSavedToRadioItem = (method: WalletInfo): RadioItem<string> | void => {
+  switch (method.details?.type) {
+    case "CARDS":
+      const cardDetails = method.details as WalletInfoDetails1;
+      return {
+        id: method.walletId,
+        value: `${capitalize(cardDetails.brand)} ••${cardDetails.maskedPan}`,
+        startImage: getIconWithFallback(cardDetails.brand)
+      };
+    case "PAYPAL":
+      return {
+        id: method.walletId,
+        value: "PayPal",
+        startImage: getIconWithFallback("paypal")
+      };
+    case "BANCOMATPAY":
+      return {
+        id: method.walletId,
+        value: "BANCOMAT Pay",
+        startImage: getIconWithFallback("bancomatpay")
+      };
+    default:
+      void null;
+  }
+};
```

Comment:
Why return void here? We should return null or undefined if no RadioItem could be returned.

```suggestion
// should never return void, but since this is a map function it's expectable
const mapSavedToRadioItem = (method: WalletInfo): RadioItem<string> | undefined => {
  switch (method.details?.type) {
    case "CARDS":
      const cardDetails = method.details as WalletInfoDetails1;
      return {
        id: method.walletId,
        value: `${capitalize(cardDetails.brand)} ••${cardDetails.maskedPan}`,
        startImage: getIconWithFallback(cardDetails.brand)
      };
    case "PAYPAL":
      return {
        id: method.walletId,
        value: "PayPal",
        startImage: getIconWithFallback("paypal")
      };
    case "BANCOMATPAY":
      return {
        id: method.walletId,
        value: "BANCOMAT Pay",
        startImage: getIconWithFallback("bancomatpay")
      };
  }
  
  return undefined;
};
```

Reply:
since this is a map function, we risk adding an `undefined` entry to a `radioItems` array. e.g.
```ts
// void null
[ {item1},{item2},{item3} ]
// undef
[{item1},undefined,{item2},{item3}]
```

Reply:
`void` is for function return type and means that the function "returns nothing". In this case, the function must return something which can be `undefined`. 

The issue with undefined items in the array can be easily managed with `fp-ts`, for example we can transform data into `Option` and then filter out the `none` items with `Array.compact`.

Reply:
I could get behind substituting the `void null` with a `break` statement, although the result would be pretty much the same; I don't see why, though, we should add complexity when it is avoidable (such as an undef filter).
Also, keep in mind that as of now the DTO only allows the three cases we are showing and `undefined`, in which case we need to not show anything.

Reply:
We should not use `void` as a return type for a function that does return a value. 
In Typescript `void` means that a function's return value will not be used, which is very different from returning `undefined`.
Marking a function that produces `undefined` with `void` could lead to type problems.

Reply:
this change was addressed when adding the new mapping logic 👍